site/profile/manifests/android.pp

```diff
@@ -0,0 +1,33 @@
+
+class profile::android (
+        String $sdk_tools,
+        String $sdk_path,
+	Array[String] $npm_packages,
+	Array[String] $sdk_packages,
+)
+
+{
+  archive { "/tmp/$sdk_tools":
+    source       => "https://dl.google.com/android/repository/$sdk_tools",
+    extract      => true,
+    extract_path => $sdk_path,
+    user          => 'ubuntu',
+    group         => 'ubuntu',
+    }
+  $npm_packages.each |String $npm_package| {
+    exec { "install_global_npm_modules_${npm_package}":
+      command  => "sudo npm i -g \"${npm_package}\"",
+    }
+  }
+  exec { "Set ANDROID_HOME":
+    command => "sed -e '/ANDROID_HOME/d' -i /etc/environment && sudo sh -c \"echo \"ANDROID_HOME=\"${sdk_path}\"\" >> /etc/environment\"",
+  }
+  $sdk_packages.each |String $sdk_package| {
+    exec { "install_android_sdk_package_${sdk_package}":
+      command => "echo y | ${sdk_path}/tools/bin/sdkmanager --install \"${sdk_package}\"",
```

Comment:
There is a command 'yes' which repeatedly echo 'y' or specific string.
It is more safe than echo yes.
So you can use something like:
yes | sdkmanager